### DISCUSSION TITLE
Cancel operation

### DISCUSSION THREAD

**Discussion by sebastianp-charactr** [icsharpcode/SharpZipLib#528] [Q&A] [answered]
Is there any way to stop ongoing zip operation while using FastZip?

**Comment by piksel** [comment, marked as answer]
Yes! Most of the FastZip events have a `ContinueRunning` property in their EventArgs that you can set to `false` to stop it. 

This is roughly how you set it up:
```cs
var fz = new FastZip(new FastZipEvents(){
    CompletedFile += (_, ea) => {
        ea.ContinueRunning = logicToDetermineIfShouldContinue()
    }
}));
```

**Reply by sebastianp-charactr** [reply]
Thanks for the reply!
The solution you've proposed is good, but my specific case is unfortunately more troublesome. Above solution will work between the files being compressed. I've got few big files to compress. Can I break the process of compressing single file in midway?

**Reply by piksel** [reply]
It should work with `Progress` event as well. That is invoked every `ProgressInterval` while compressing:
```cs
var fz = new FastZip(new FastZipEvents(){
    Progress+= (_, ea) => {
        ea.ContinueRunning = logicToDetermineIfShouldContinue()
    },
    ProgressInterval = TimeSpan.FromSeconds(1),
}));
```

**Reply by sebastianp-charactr** [reply]
Great, thanks for your help!
